cmd/bucket-handlers.go

```diff
@@ -197,7 +199,17 @@ func (api objectAPIHandlers) ListBucketsHandler(w http.ResponseWriter, r *http.R
 	}
 
 	// List buckets does not support bucket policies, no need to enforce it.
-	if s3Error := checkAuth(r); s3Error != ErrNone {
+	// Proceed to validate signature.
```

Comment:
errorIf is not useful since we are going to send back AccessDenied in such a case. Let me making a readable function.